crates/libp2p-networking/src/network/node/handle.rs

```diff
@@ -156,15 +152,10 @@ impl NetworkNodeHandle {
         &self,
         num_peers: usize,
         node_id: usize,
-        timeout: Duration,
     ) -> Result<(), NetworkNodeHandleError> {
-        let start = Instant::now();
         self.begin_bootstrap().await?;
         let mut connected_ok = false;
         while !connected_ok {
-            if start.elapsed() >= timeout {
-                return Err(NetworkNodeHandleError::ConnectTimeout);
-            }
```

Comment:
Before this, it would return an error but the caller would panic on it. I think there may be a more elegant solution to this, one where we can time out on the caller's caller (the sequencer or examples) instead and have them do logic with it. It makes sense for the logic to be there since we can do something conditionally with the CDN network. Let me think on this

Reply:
Hmm, I think this solution is alright. Instead of timing out after 10 minutes it will try forever. I'm going to improve it by holding a handle to this task and killing it if we drop the Libp2p network. This way, upstream in the sequencer we can have better cancellation logic